safe/load.go

```diff
@@ -15,19 +17,45 @@ func Load(password []byte, backend backends.Client, encryptionClient crypto.Clie
 		password: password,
 	}
 
-	ciphertext, err := s.backend.Load()
-
-	if _, ok := err.(*errors.SafeNotFound); ok {
-		s.Accounts = make(map[string]Account)
-		return &s, nil
-	}
+	data, err := s.backend.Load()
 	if err != nil {
+		if _, ok := err.(*errors.SafeNotFound); ok {
+			s.Accounts = make(map[string]Account)
+			return &s, nil
+		}
 		return nil, err
 	}
 
-	plaintext, err := s.crypto.Decrypt(ciphertext, password)
+	safeFile := safeFile{}
+	if err := json.Unmarshal(data, &safeFile); err == nil {
+		// Unmarshalling did succeed -> Safe uses new format
+		// Do nothing
+	} else {
+		safeFile.Ciphertext = data
+		// If we don't have a config, use OpenPGP for backwards-compatibility
+		defaultOpenPGPConfig := crypto.DefaultConfigWithType("aes_openpgp")
+		safeFile.Config = &defaultOpenPGPConfig
+	}
+
+	upgradeNeeded := false
+	plaintext, err := s.crypto.Decrypt(safeFile.Ciphertext, password)
```

Comment:
Was not able to reproduce. Works fine for me. What exactly did you change in step `4.`?

How I tried it:

1. Remove any existing `pick.safe` and `config.toml`
2. `$ pick add test1 test1 test1`
3. Set master password to `pswd`
4. `make config`
5. `$ vim ~/.pick/config.toml`
5.1. Uncomment `[encryption]`
5.2. Uncomment `type = "aes_gcm"`
5.3. Uncomment `[encryption.settings.aes_gcm]`
5.4. Uncomment `keylen = 32`
5.5. Set `keylen = 24`
5.6. Save the config
6. `$ pick cat test1` / `$ pick add test2 test2 test2`
7. "Upgrading safe"
8. `$ pick cat test1` / `$ pick cat test2`
9. Works fine.